src/main/java/frc/robot/subsystems/CrocodileSubsystem.java

```diff
@@ -1,92 +1,68 @@
+    private final WPI_TalonFX wrist;
+    private final PIDController pid;
+    private final DutyCycleEncoder encoder;
+    private final DigitalInput beamBreak;
+
+    //TODO: Set speed limit
+    private final double SPEED_LIMIT = 0;
 
     public CrocodileSubsystem() {
         intake = new WPI_TalonFX(RobotMap.Crocodile.INTAKE);
+        wrist = new WPI_TalonFX(RobotMap.Crocodile.WRIST);
+        pid = new PIDController(0, 0, 0);
+        encoder = new DutyCycleEncoder(1);
+        beamBreak = new DigitalInput(2);
         intake.setNeutralMode(NeutralMode.Brake);
+        wrist.setNeutralMode(NeutralMode.Brake);
         /* enabled | Limit(amp) | Trigger Threshold(amp) | Trigger Threshold Time(s) */
         intake.configSupplyCurrentLimit(new SupplyCurrentLimitConfiguration(true, 35, 60, 0.1));
+        setSetpoint(encoder.getAbsolutePosition());
 
-        wristSolenoid = new Solenoid(
-                RobotMap.Pneumatics.PCM,
-                RobotMap.Pneumatics.PCM_TYPE,
-                RobotMap.Pneumatics.CLAW_PCM_CHANNEL);
-        chomperSolenoid = new Solenoid(
-                RobotMap.Pneumatics.PCM,
-                RobotMap.Pneumatics.PCM_TYPE,
-                RobotMap.Pneumatics.CHOMPER_PCM_CHANNEL);
     }
 
-    public void setMotorSpeed(double speed) {
+    public void setIntakeSpeed(double speed) {
         intake.set(speed);
-        currentMotorPower = speed;
-    }
-
-    public double getMotorVelocity() {
-        return intake.getSelectedSensorVelocity();
-    }
-
-    private void setWrist(boolean newValue) {
-        wristSolenoid.set(newValue);
-    }
-
-    public void toggleWrist() {
-        wristSolenoid.toggle();
-    }
-
-    public boolean getWristPositionBool() {
-        return wristSolenoid.get();
-    }
-
-    public void wristUp() {
-        setWrist(true);
-    }
-
-    public void wristDown() {
-        setWrist(false);
     }
 
-    private void setChomper(boolean newValue) {
-        chomperSolenoid.set(newValue);
+    private void setWristSpeed(double speed) {
+        wrist.set(speed);
     }
 
-    public void toggleChomper() {
-        chomperSolenoid.toggle();
+    public void setSetpoint(double position) {
+        pid.setSetpoint(position);
     }
 
-    public void closeChomper() {
-        setChomper(true);
+    public double getSetpoint() {
+        return pid.getSetpoint();
     }
 
-    public void openChomper() {
-        setChomper(false);
+    public double getAbsolutePosition() {
+        return encoder.getAbsolutePosition();
     }
 
-    public boolean getChomperPositionBool() {
-        return chomperSolenoid.get();
+    public boolean getBeamBreak() {
+        return beamBreak.get();
     }
 
     @Override
     public void periodic() {
-        // if (currentMotorPower > 0 && stallDebouncer.calculate((getMotorVelocity() <
-        // 50))) {
-        // setMotorSpeed(0);
-        // }
+        setWristSpeed(pid.calculate(encoder.getAbsolutePosition()) * SPEED_LIMIT);
```

Comment:
Good, thank you for adding a speed limit! Why is it a multiplier instead of a clamp? I would prefer a clamp, because if the PID decides to set the power to an obscene value, which is what we want to protect against, a multiplier will do very little. Clamping the value makes sure it never goes beyond or below a value, so it is much more reliable and safe.